vcs/gitcmd/repo.go

```diff
@@ -890,6 +901,45 @@ func (r *Repository) Search(at vcs.CommitID, opt vcs.SearchOptions) ([]*vcs.Sear
 	return res, err
 }
 
+func (r *Repository) Committers(opt vcs.CommittersOptions) ([]*vcs.Committer, error) {
+	r.editLock.RLock()
+	defer r.editLock.RUnlock()
+
+	if opt.Rev == "" {
+		opt.Rev = "HEAD"
+	}
+
+	cmd := exec.Command("git", "shortlog", "-sne", opt.Rev)
+	cmd.Dir = r.Dir
+	out, err := cmd.CombinedOutput()
+	if err != nil {
+		return nil, fmt.Errorf("exec `git shortlog -sne` failed: %v. Output was:\n\n%s", err, string(out))
+	}
+	out = bytes.TrimSpace(out)
```

Comment:
You are using `cmd.CombinedOutput()` (which includes stdout and stderr) to get the value of `out` here, and then try to parse it.

There are two reasons to avoid doing that.

1. Stderr should not be necessary. The output you're looking to parse should be completely contained within `stdout`.

2. It can cause issues. Git (and other commands, in general) use stderr to output optional warning messages and the like. If those are present, it will likely cause parsing to fail or return invalid results. See issue #30 where it actually happened and related issue #33 that's still open.

I highly suggest using `cmd.Output()` to capture the value of stdout only, and parsing that. Stderr should not be needed at all, as it's only useful for reporting an error when it occurs. But I don't think it's important enough to spend extra effort capturing it.